api/tests/Feature/NotificationTest.php

```diff
@@ -0,0 +1,467 @@
+            ->graphQL($this->queryNotifications, [
+                'where' => [
+                    'onlyUnread' => true,
+                ],
+            ]);
+
+        $responseNotifications = $response->json('data.notifications.data');
+        $this->assertCount(count($unreadNotifications), $responseNotifications);
+
+        foreach ($responseNotifications as $shouldBeUnread) {
+            $this->assertNull($shouldBeUnread['readAt']);
+        }
+    }
+
+    public function testDateRangeFilter()
+    {
+        $this->user->notify(
+            new PoolCandidateStatusChanged(
+                PoolCandidateStatus::PLACED_INDETERMINATE->name,
+                PoolCandidateStatus::QUALIFIED_AVAILABLE->name,
+                $this->pool->id,
+                $this->pool->name
+            )
+        );
+
+        $pastDate = new Carbon(config('constants.past_date'));
+        $this->user->notifications()->first()->update([
+            'read_at' => $pastDate,
+        ]);
+        $this->user->unreadNotifications->markAsRead();
+
+        $from = $pastDate->addDay();
+        $response = $this->actingAs($this->user, 'api')
+            ->graphQL($this->queryNotifications, [
+                'where' => [
+                    'readAt' => [
+                        'from' => $from->toDateTimeString(),
+                        'to' => Carbon::now()->toDateTimeString(),
+                    ],
+                ]]);
+
+        $notifications = $this->user->notifications()->get()->toArray();
+
+        $rangeResponse = $response->json('data.notifications.data');
+
+        $this->assertCount(count($notifications) - 1, $rangeResponse);
+
+        foreach ($rangeResponse as $shouldBeInRange) {
+            $date = new Carbon($shouldBeInRange['readAt']);
+
+            $this->assertTrue($date->gt($from));
+
+        }
+    }
+
+    /**
+     * Test that notifications are generated for status changes
+     *
+     * @return void
+     */
+    public function testStatusChangeCausesNotifications()
+    {
+        if (! config('feature.status_notifications')) {
+            $this->markTestSkipped('This test uses features behind the FEATURE_STATUS_NOTIFICATIONS flag.');
+        }
+
+        NotificationFacade::fake(); // initialize notification facade
+
+        // set up submitted candidate
+        $this->poolCandidate->pool_candidate_status = PoolCandidateStatus::NEW_APPLICATION->name;
+        $this->poolCandidate->saveQuietly(); // don't generate events during set up
+
+        // simulate screening in
+        $this->poolCandidate->pool_candidate_status = PoolCandidateStatus::SCREENED_IN->name;
+        $this->poolCandidate->save();
+
+        // check that notification was fired
+        NotificationFacade::assertSentTo([$this->candidateUser], PoolCandidateStatusChanged::class);
+    }
+
+    /**
+     * Test that use can query for notifications
+     *
+     * @return void
+     */
+    public function testCanQueryForNotifications()
```

Comment:
This one, I'm not sure if it should be moved to the new test file or just be deleted. Does it test anything above and beyond the simpler tests in this file? 

Maybe it was for testing that the "enrich notification" method worked? Do we need to test that anymore?

Reply:
Hmm, yeah it is a little more comprehensive and strict by actually checking the fields exist and match. I guess I could move the assertions to the other query test and then delete this one?

Reply:
I like it!